File: apps/customer_mobile/lib/core/utils/restaurant_card_meta.dart
import '../../shared/models/restaurant_model.dart';

int restaurantPrepMinutes(RestaurantModel r) {
  return r.deliveryMinutes ?? r.avgPrepMinutes ?? 30;
}

String restaurantDeliveryLabel(RestaurantModel r) {
  final base = restaurantPrepMinutes(r);
  final low = (base - 5).clamp(15, 999);
  final high = base + 5;
  return '$low – $high daq';
}

String restaurantCategoryLabel(RestaurantModel r) {
  final cats = r.categories ?? [];
  return cats.map((c) => c.name.trim()).where((n) => n.isNotEmpty).join(', ');
}

String? restaurantRatingLabel(RestaurantModel r) {
  final rating = r.averageRating;
  if (rating == null) return null;
  final value = rating.toStringAsFixed(1);
  final count = r.reviewCount ?? 0;
  return '★ $value ($count)';
}

File: frontend/src/lib/restaurant-card-meta.ts
import { uz } from '@/lib/uz';

type CategoryLike = { name: string };

type RestaurantMeta = {
  avgPrepMinutes?: number;
  deliveryMinutes?: number;
  averageRating?: number;
  reviewCount?: number;
  categories?: CategoryLike[];
  productCategories?: CategoryLike[];
};

export function restaurantPrepMinutes(r: RestaurantMeta): number {
  return r.deliveryMinutes ?? r.avgPrepMinutes ?? 30;
}

export function restaurantDeliveryLabel(r: RestaurantMeta): string {
  const base = restaurantPrepMinutes(r);
  const low = Math.max(15, base - 5);
  const high = base + 5;
  return uz.deliveryTime(`${low} – ${high} ${uz.min}`);
}

export function restaurantCategoryLabel(r: RestaurantMeta): string {
  const cats = r.productCategories ?? r.categories ?? [];
  return cats
    .map((c) => c.name.trim())
    .filter(Boolean)
    .join(', ');
}

export function restaurantRatingLabel(r: RestaurantMeta): string | null {
  const rating = r.averageRating;
  if (rating == null || Number.isNaN(Number(rating))) return null;
  const value = Number(rating).toFixed(1);
  const count = r.reviewCount ?? 0;
  return `★ ${value} (${count})`;
}
